### ISSUE TITLE
Filename highlighting not completely working when filtering

### ISSUE THREAD

**Issue by fritzrehde** [sxyazi/yazi#497]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

Alacritty 0.13.1

### Yazi version

yazi 0.1.5 (c3a8bdc 2024-01-09)

### Did you try the latest main branch to see if the problem has already been fixed?

Tried, but the problem is still present

### Describe the bug

When I have a directory with more files/directories inside than fit on the current screen, filtering for file/dir names causes those files/dirs that were previously not displayed on the current screen to *not* be highlighted. See this demo video (the left part is chopped off for privacy reasons, I hope the bug is still visible. In the bottom left of the screen, I have the `filter` popup window where I am typing `.mp4`):

https://github.com/sxyazi/yazi/assets/80471265/f76f23da-58e3-434d-8e70-b447efa815aa

 

### Expected Behavior

I would expect the newly shown file/dirs (those that were newly filtered out) to also have highlighting (i.e. be colored yellow like the originally visible files/dirs).

### To Reproduce

I hope I described enough in the "Describe the bug" section. If not, please let me know.

### Configuration

no config

**Comment by fritzrehde** [comment]
I assume this bug is due to some optimization where only the files/dirs visible on screen should get highlighted, but then for this specific case somewhere there is a notification missing that the newly filtered files/dirs, that were previously not shown on screen, are now visible.

**Comment by sxyazi** [comment]
Hi, I made a PR to fix this, https://github.com/sxyazi/yazi/pull/505, let me know if it works for you!

For better performance, it won't update during the filtering process (with each key press) - this would lead to pre-caching a large number of images and videos for multiple "temporary results".

Instead, it only triggers an update upon completion of the filtering - by pressing `Enter` on the input. I found that this approach works well and is also intuitive.
